core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -177,6 +177,8 @@ private[spark] class DAGScheduler(
   // TODO: Garbage collect information about failure epochs when we know there are no more
   //       stray messages to detect.
   private val failedEpoch = new HashMap[String, Long]
```

Comment:
I suggest we rename this to `blockManagerFailureEpoch`, or perhaps `executorFailureEpoch` and change the comment significantly, to something like

Tracks the latest epoch of a fully processed error related to the given executor.

When an executor fails, it will affect the results of many tasks, and we have to deal with all of them consistently.  We don't simply ignore all future results from that executor, as the failures may have been transient; but we also don't want to "over-react" to the many follow-on errors we'll receive.  Furthermore, we might receive notification of a task success, after we find out the executor has actually failed; we'll assume those successes are, in fact, simply delayed notifications and the results have been lost, if they come from the same epoch.  In particular, we use this to control when we tell the blockManagerMaster that the blockManager has been lost.

Reply:
I changed `failedEpoch` to `executorFailureEpoch` and more or less adopted your suggestion for the comment explaining it.